include/drivers/video.h

```diff
@@ -326,15 +327,15 @@ static inline int video_enqueue(struct device *dev, enum video_endpoint_id ep,
  * @param dev Pointer to the device structure for the driver instance.
  * @param ep Endpoint ID.
  * @param buf Pointer a video buffer pointer.
- * @param timeout Timeout in milliseconds.
+ * @param timeout Timeout
  *
  * @retval 0 Is successful.
  * @retval -EINVAL If parameters are invalid.
  * @retval -EIO General input / output error.
  */
 static inline int video_dequeue(struct device *dev, enum video_endpoint_id ep,
 				struct video_buffer **buf,
-				u32_t timeout)
+				k_timeout_t timeout)
```

Comment:
Note that this is part of the video API which is marked as "stable".

Reply:
You are very right. I will undo this change I think, and just leave the current API as-is.

Reply:
I don't care much about the video API per se, but in fact the magic of CONFIG_LEGACY_TIMEOUT_API is that if you really need backwards source compatibility, you have it even with k_timeout_t.  So applications that need the stable API don't lose it, even if they need the new features.

Whether or not we care about video dequeue timers being high precision is sort of an open question though.  I'd expect they'd be happy with latency near ~30ms video frame time and wouldn't care.

Reply:
Actually, I will leave the change but add a note in the release notes and run it by everybody in the API meeting

Reply:
Yeah, honestly the main reason I switched to `k_timeout_t` is that this was `u32_t` in the first place, so it was wrong. I could've just as easily switched to `s32_t` for consistency with other driver APIs, perhaps it's just better if I do that?